rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,229 @@
+package co.rsk.peg;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.core.RskAddress;
+import co.rsk.bitcoinj.store.BlockStoreException;
+import co.rsk.net.utils.TransactionUtils;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.peg.btcLockSender.BtcLockSenderProvider;
+import co.rsk.peg.constants.BridgeConstants;
+import co.rsk.peg.constants.BridgeMainNetConstants;
+import co.rsk.peg.federation.*;
+import co.rsk.peg.federation.constants.FederationConstants;
+import co.rsk.peg.feeperkb.FeePerKbSupport;
+import co.rsk.peg.storage.BridgeStorageAccessorImpl;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import co.rsk.test.builders.FederationSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.*;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+
+public class RegisterBtcTransactionIT {
+
+    private final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    private final NetworkParameters btcParams = bridgeConstants.getBtcParams();
+    private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+
+    @Test
+    void whenRegisterALegacyBtcTransaction_shouldRegisterTheNewUtxoAndTransferTheRbtcBalance() {
+        // Arrange
+        ActivationConfig.ForBlock activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        Repository repository = BridgeSupportTestUtil.createRepository();
+        Repository track = repository.startTracking();
+        Block rskExecutionBlock = getRskExecutionBlock();
+
+        BtcLockSenderProvider btcLockSenderProvider = new BtcLockSenderProvider();
+        FeePerKbSupport feePerKbSupport = getFeePerKbSupport();
+
+        Federation federation = P2shErpFederationBuilder.builder().build();
+        FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
+        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
```

Comment:
Should this not be in `setup` in the `beforeTest` method since I guess you will need it for other methods, right?

Reply:
Yep! I left it here because there was just one test. For the following ones I added the setUp method with @BeforeEach